crates/re_arrow_store/tests/dump.rs

```diff
@@ -0,0 +1,236 @@
+//! Dumping a datastore to log messages and back.
+
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use itertools::Itertools;
+use re_arrow_store::{test_row, DataStore, DataStoreStats, TimeInt, TimeRange, Timeline};
+use re_log_types::{
+    component_types::InstanceKey,
+    datagen::{
+        build_frame_nr, build_log_time, build_some_colors, build_some_instances, build_some_point2d,
+    },
+    Component as _, DataTable, EntityPath, TableId,
+};
+
+// --- Dump ---
+
+#[test]
+fn data_store_dump() {
+    init_logs();
+
+    for mut config in re_arrow_store::test_util::all_configs() {
+        // NOTE: insert IDs aren't serialized and can be different across runs.
+        config.store_insert_ids = false;
+
+        let mut store1 = DataStore::new(InstanceKey::name(), config.clone());
+        let mut store2 = DataStore::new(InstanceKey::name(), config.clone());
+        let mut store3 = DataStore::new(InstanceKey::name(), config.clone());
+
+        data_store_dump_impl(&mut store1, &mut store2, &mut store3);
+    }
+}
+
+fn data_store_dump_impl(store1: &mut DataStore, store2: &mut DataStore, store3: &mut DataStore) {
```

Comment:
Why pass in the three stores by `&mut ` when they must start out empty, and are not used outside the function?

```suggestion
        data_store_dump_impl(config);
    }
}

fn data_store_dump_impl(config) {
    let mut store1 = DataStore::new(InstanceKey::name(), config.clone());
    let mut store2 = DataStore::new(InstanceKey::name(), config.clone());
    let mut store3 = DataStore::new(InstanceKey::name(), config.clone());
}
```

Reply:
1. Consistency with other test suites
2. I like keeping initialization and config permutation logic separate from the data test